test/Models/CsvCatalogItemSpec.js

```diff
@@ -94,31 +111,318 @@ describe('CsvCatalogItem', function() {
-        expect(csvItem instanceof CatalogItem).toBe(true);
+    it('identifies "lat" and "lon" fields', function(done) {
+        csvItem.updateFromJson( { data: 'lat,lon,value\n-37,145,10' });
+        csvItem.load().then(function() {
+            expect(csvItem._tableDataSource.dataset.hasLocationData()).toBe(true);
+        }).otherwise(fail).then(done);
+    });
+    it('identifies "latitude" and "longitude" fields', function(done) {
+        csvItem.updateFromJson( { data: 'latitude,longitude,value\n-37,145,10' });
+        csvItem.load().then(function() {
+            expect(csvItem._tableDataSource.dataset.hasLocationData()).toBe(true);
+        }).otherwise(fail).then(done);
+    });
+    it('does not mistakenly identify "latvian" and "lone_person" fields', function(done) {
+        csvItem.updateFromJson( { data: 'latvian,lone_person,lat,lon,value\n-37,145,-37,145,10' });
+        csvItem.load().then(function() {
+            expect(csvItem._tableDataSource.dataset.getVariableNamesByType(VarType.LON)).toEqual(['lon']);
+            expect(csvItem._tableDataSource.dataset.getVariableNamesByType(VarType.LAT)).toEqual(['lat']);
+        }).otherwise(fail).then(done);
+    });
+    it('handles numeric fields containing (quoted) thousands commas', function(done) {
+        csvItem.updateFromJson( { data: 'lat,lon,value\n-37,145,"1,000"\n-38,145,"234,567.89"' });
+        csvItem.load().then(function() {
+            expect(csvItem._tableDataSource.dataset.hasLocationData()).toBe(true);
+            expect(csvItem._tableDataSource.dataset.getDataValue('value', 0)).toEqual(1000);
+            expect(csvItem._tableDataSource.dataset.getDataValue('value', 1)).toBeCloseTo(234567.89,2);
+        }).otherwise(fail).then(done);
     });
 
-    it('returns an error on non-csv text', function() {
-        expect(csvItem instanceof CatalogItem).toBe(true);
+    it('matches LGAs by code', function(done) {
+        csvItem.updateFromJson( { data: 'lga_code,value\n31000,1' });
+        csvItem.load().then(function() {
+            expect(csvItem._regionMapped).toBe(true);
+            expect(csvItem.colorFunc).toBeDefined();
+            // 242 is the shapefile index of LGA boundary 31000. What a crappy way to test...
+            expect(csvItem.colorFunc(242)).not.toEqual([0,0,0,0]);
+        }).otherwise(fail).then(done);
+
     });
+    it('matches LGAs by names in various formats', function(done) {
+        csvItem.updateFromJson( { data: 'lga_name,value\nCity of Melbourne,1\nGreater Geelong,2\nSydney (S),3' });
+        csvItem.load().then(function() {
+            expect(csvItem._regionMapped).toBe(true);
+            expect(csvItem.colorFunc).toBeDefined();
+            expect(csvItem.colorFunc(121)).not.toEqual([0,0,0,0]);
+            expect(csvItem.colorFunc(180)).not.toEqual([0,0,0,0]);
+            expect(csvItem.colorFunc(197)).not.toEqual([0,0,0,0]);
+        }).otherwise(fail).then(function() { done(); });
```

Comment:
Right. I was [going a little crazy](http://stackoverflow.com/questions/32431544/idiomatically-testing-when-js-promises-with-jasmine/32433098?noredirect=1#comment52730976_32433098) trying to get Jasmine and (our version of) When to play together.